pkg/image/api/helper.go

```diff
@@ -307,6 +307,38 @@ func AddTagEventToImageStream(stream *ImageStream, tag string, next TagEvent) bo
 	return true
 }
 
+// UpdateChangedTrackingTags identifies any tags in the status that have changed and
+// ensures any referenced tracking tags are also updated. It returns the number of
+// updates applied.
+func UpdateChangedTrackingTags(new, old *ImageStream) int {
+	changes := 0
+	for newTag, newImages := range new.Status.Tags {
+		if oldImages, ok := old.Status.Tags[newTag]; ok {
+			changed, deleted := tagsChanged(oldImages.Items, newImages.Items)
+			if !changed || deleted {
```

Comment:
Kind of long how?

On Oct 21, 2015, at 4:08 AM, Michal Minar <notifications@github.com> wrote:

In pkg/image/api/helper.go
<https://github.com/openshift/origin/pull/5267#discussion_r42593988>:

> @@ -307,6 +307,38 @@ func AddTagEventToImageStream(stream *ImageStream, tag string, next TagEvent) bo
>  	return true
>  }
>
> +// UpdateChangedTrackingTags identifies any tags in the status that have changed and
> +// ensures any referenced tracking tags are also updated. It returns the number of
> +// updates applied.
> +func UpdateChangedTrackingTags(new, old *ImageStream) int {
> +	changes := 0
> +	for newTag, newImages := range new.Status.Tags {
> +		if oldImages, ok := old.Status.Tags[newTag]; ok {
> +			changed, deleted := tagsChanged(oldImages.Items, newImages.Items)
> +			if !changed || deleted {

Wouldn't

if len(newImages.Items) > 0 && (len(oldImages.Items) == 0 ||
oldImages.Items[0] != newIMages.Items[0]) {
    changes += UpdateTrackingTags(new, newTag, newImages.Items[0])
}

with some comment suffice? tagsChanged() is too specific to be used
elsewhere and kind of long.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5267/files#r42593988>.